### PR TITLE
Some API changes to blob service

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:9]:
```suggestion
- Removed source lease ID access conddition for `BlobClient::StartCopy()`, because it's never supported.

### New Features
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:8]:
```suggestion
- Removed source lease ID access conditions for `BlobClient::StartCopy()`, because it's never supported.
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (antkmsft) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:5]: Minor version releases can never have breaking changes. In order to do what you are doing, you need to release version `13.0.0-beta.1`.

Also, from now on, we use new headings - https://github.com/Azure/azure-sdk-for-cpp/pull/2527#discussion_r662516141 (which we don't apply for the releases already made)

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (antkmsft) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:16]:
```suggestion
### Bugs Fixed
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (antkmsft) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:10]:
```suggestion
### Features Added
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (antkmsft) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:5]: What you can do, is to leave the old things as they were, but ignore them (most likely documenting that they have no effect).
So, the client code someone had with `12.0.0` should still compile and behave the same.

**Review by antkmsft — CHANGES_REQUESTED** [review]
Please remove the code breaking changes.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md
Comment (katmsft) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:5]: Would need approval from Vincent for this.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md]: reverted

**Comment by Jinming-Hu** [comment]
blocked by https://github.com/Azure/azure-sdk-for-cpp/pull/2558

**Comment by Jinming-Hu** [comment]
> Please remove the breaking changes (what would break compilation, and/or program behavior for the previous version user).

Reverted

**Review by antkmsft — APPROVED** [review]

**Pull request merged**
